azurerm/internal/services/apimanagement/api_management_gateway_data_source.go

```diff
@@ -0,0 +1,87 @@
+package apimanagement
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/apimanagement/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/apimanagement/schemaz"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceApiManagementGateway() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceApiManagementGatewayRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": schemaz.SchemaApiManagementChildDataSourceName(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"api_management_name": schemaz.SchemaApiManagementDataSourceName(),
```

Comment:
Can we use `api_management_id` here instead or `api_management_id` and `resource_group_name`? (See https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/internal/services/apimanagement/api_management_custom_domain_resource.go#L39-L44 for example implementation)